### PR TITLE
Feature/SOF-7286

### PR DESCRIPTION
**Pull request by VsevolodX**
Restructures JupyterLite session component to use JLS component from cove.js

### CODE REVIEW

File: src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx [hunk 1]
```diff
@@ -41,7 +44,9 @@ class JupyterLiteTransformation extends React.Component<
     }
 
     componentDidMount() {
-        window.addEventListener("message", this.handleReceiveMessage, false);
+        this.messageHandler.addHandlers("set-data", [this.setMaterials]);
+        this.messageHandler.addHandlers("get-data", [this.returnSelectedMaterials]);
+        this.messageHandler.sendData(this.returnSelectedMaterials());
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx]: This should be requested by JupyterLite after it's loaded

File: src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx [hunk 2]
```diff
@@ -50,33 +55,24 @@ class JupyterLiteTransformation extends React.Component<
...
-                    });
-                }
-            } catch (err) {
-                console.log(err);
-            }
-            if (event.data.requestData === true && event.data.variableName === "materials_in") {
-                this.sendMaterialsToIFrame();
+    setMaterials = (data: any) => {
+        try {
+            const configs = data.materials as MaterialSchema[];
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx]: Let's add a comment explaining why try/catch is needed here

File: src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx [hunk 3]
```diff
@@ -50,33 +54,29 @@ class JupyterLiteTransformation extends React.Component<
...
-                this.sendMaterialsToIFrame();
+    setMaterials = (data: any) => {
+        // Since any arbitrary data can be sent from the JupyterLite notebook, the validation might fail
+        try {
+            const configs = data.materials as MaterialSchema[];
+            if (Array.isArray(configs)) {
+                this.setState({
+                    newMaterials: configs.map((config) => {
+                        const m = new Made.Material(config);
+                        m.validate();
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx]: Let's show an error message to the user?? Using notistack.

The logic should be:

1. Run validation for all configs
2. Throw exception and show user a message explaining which item is incorrect and why
3. Filter out the configs that don't pass validation
4. Proceed with the rest of the configs to set the state

File: src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx [hunk 4]
```diff
@@ -50,33 +55,38 @@ class JupyterLiteTransformation extends React.Component<
...
-        }
-        if (event.data.type === "from-iframe-to-host") {
-            try {
-                // TODO: add check for Material config type
-                const configs = event.data.data.materials;
-                if (Array.isArray(configs)) {
-                    this.setState({
-                        newMaterials: configs.map((config) => new Made.Material(config)),
-                    });
+    setMaterials = (data: any) => {
```
Comment (timurbazhirov) [inline src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx]: set materials should just set materials with setState, this is much more complicated function. Should be called handleSetData or something like that

Comment (VsevolodX) [reply src/components/3d_editor/advanced_geometry/python_transformation/JupyterLiteTransformation.tsx]: Ok

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**